posthog/models/filters/mixins/property.py

```diff
@@ -11,7 +11,7 @@
 
 class PropertyMixin(BaseParamMixin):
     @cached_property
-    def properties(self) -> List[Property]:
+    def properties(self) -> Union[List[Property], List[List[Property]]]:
```

Comment:
This PR will come down to typing making sense IMO.

Let's rename current properties to something like `VanillaProperty`.

I think `List[AnyProperty]` is the proper value here, with `AnyProperty = Union[VanillaProperty, OrProperty]`

OrProperty should be a separate class from the property containing neccessary values.

How I imagine the class will look like:

```
class OrProperty:
  type = 'or'

  def __init__(properties: List[AnyProperty]): # ... 
```

From the frontend you'd pass a payload like `properties: [{ type: 'or', properties: [...] }]`

Reply:
Hmm i see, will give that a shot when i loop back around to this